crates/bifrost/src/service.rs

```diff
@@ -83,7 +83,10 @@ impl BifrostService {
         // Make sure we have v1 metadata written to metadata store with the default
         // configuration. If metadata is already initialized, this will make sure we have the
         // latest version set in metadata manager.
-        self.bifrost.admin().init_metadata().await?;
+
+        // todo we seem to have a race condition between this call and the provision step which might
+        //  write a different logs configuration
+        // self.bifrost.admin().init_metadata().await?;
```

Comment:
This problem should already exist in the current main. The problem is that we first provision the `NodesConfiguration` and then try to write the `Logs` with the configured `LogsConfiguration`. If now the node succeeds at joining the cluster and reaches first the point where it starts the `BifrostService`, then it can happen that this line writes the initial `Logs` which does not respect the `LogsConfiguration` that is specified in the provision command.